### ISSUE TITLE
stylus support

### ISSUE THREAD

**Issue by prog666** [keystonejs/keystone#1314]
it asks less or sass, but i want stylus! =)

**Comment by morenoh149** [comment]
that PR is still welcome https://github.com/keystonejs/keystone/issues/183

**Comment by tmwalton** [comment]
It looks like Stylus is supported when I setup a Keystone project, but the css doesn't work. I know this ticket is closed, but the comments leave the status unclear. Is Stylus supported or not?

**Comment by morenoh149** [comment]
@tmwalton what do you mean stylus is supported but the css doesn't work?

**Comment by tmwalton** [comment]
What I mean is that if I select Stylus when setting up my project, run "node keystone", and look at localhost:3000, there are no css styles being applied to the html in the browser.
If I create a project and select Less, and do exactly the same, the css is applied to the html appropriately.
I spent some time fiddling with my project but I was unable to make it work with Stylus.

**Comment by PimpShanti** [comment]
Seconded. When you select Stylus in the Yeoman generator you end up with a site with no css file.

I've only just started learning JavaScript, node.js etc and so naturally I also figured I must have done something wrong, but after 2 days of wrestling with all these concepts I have realised that the Yeoman generator is creating a setup that defines Bootstrap with Stylus and no sign of a Stylus compiler (although maybe I'm still missing that??). Certainly when Keystone is running, there is no css file in public/styles, just a 'site.styl' file.

 To my mind, Stylus is not yet properly supported and it's inclusion in the Yeoman Generator breaks stuff.

**Comment by jkallunki** [comment]
I'm using Keystone for the first time and had this issue as well. I use Gulp and Stylus in my project. The project has to be started with `gulp` instead of `node keystone` to get the watch:stylus Gulp task running. In addition, that doesn't compile the .styl files instantly, but only after the files are actually changed when the task is running.

**Comment by morenoh149** [comment]
@jkallunki please adds docs https://github.com/keystonejs/keystonejs-site
or add a note to the generator output https://github.com/keystonejs/generator-keystone

**Comment by jkallunki** [comment]
It might be a good idea to modify the generator and docs so that the output project should always be started with `npm start`, so that it always runs the correct task (gulp or grunt) according to the generator input. I haven't tried using Grunt with Keystone yet, but I guess that there might be similar issues. I'll try it later and make a pull request according to my findings. What do you think? @morenoh149

**Comment by morenoh149** [comment]
definitely

**Comment by jkallunki** [comment]
Now when I'm a little more familiar with the codebase, I suggest adding Stylus middleware binding to the server, so that the .styl files will be compiled when the server is started, even when Grunt and Gulp are not used. That would be in line with the bindSassMiddleware.js and bindLessMiddleware.js already found in keystone/server. I don't know if that has been tried already at some point, but I can check it tomorrow and make a PR to this repository if it works. Should this issue be re-opened for that?

When the server does the initial compilation (both in local dev env and production), Grunt and Gulp can be used locally to watch changes and re-compile styles on the fly. For that I'd make another pull request that closes https://github.com/keystonejs/generator-keystone/issues/147

**Comment by morenoh149** [comment]
@jkallunki I think it's better to make another issue for that
